packages/cache/src/internal/downloadUtils.ts

```diff
@@ -247,7 +249,12 @@ export async function downloadCacheStorageSDK(
 
     try {
       downloadProgress.startDisplayTimer()
-
+      const abortTimeInMs =
+        options.abortTimeInMs === undefined ? 3600000 : options.abortTimeInMs
+      const abortSignal = AbortController.timeout(abortTimeInMs)
+      abortSignal.addEventListener('abort', () => {
+        core.warning('Cache download aborted, segment download timed out.')
```

Comment:
Don't we have to throw error here or will `downloadToBuffer` throw? If latter then do we really need to log warning here or will the error contains timeout information?

Reply:
Throwing new error here [stops the workflow with errors](https://github.com/bbq-beets/test-cache-load/runs/7689586482?check_suite_focus=true#step:2:34). Hence just throwing warning to let the rest of the workflow [succeed, but with warnings](https://github.com/bbq-beets/test-cache-load/actions/runs/2803164508)

Reply:
@kotewar , that error will be catched here right https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L129-L136 and set as warning?
THrowing error might be useful here so that partial download dont get untarred [here](https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L125)

Reply:
Its not getting caught as a warning, instead the execution is stopped. Please check [this output](https://github.com/bbq-beets/test-cache-load/runs/7689586482?check_suite_focus=true#step:2:34)

Reply:
@tiwarishub I tested by added a new debug log in my repo - https://github.com/kotewar/cache/blob/main/src/restore.ts#L48

This log should have printed in the workflow file, but since the `cacheKey` is not found, the restore run is exiting there. Check [this run](https://github.com/bbq-beets/test-cache-load/runs/7719216695?check_suite_focus=true#step:2:36)